rfc/custom-modules-v1.md

```diff
@@ -332,21 +360,56 @@ The new module system is not backwards compatible with the old module system. Ex
 
 __All examples are pseudocode and not tested, but they are as close as possible to the final implementation__
 
-## Custom Telemetry
+## Custom Metrics
 
-This module adds custom attributes to the OpenTelemetry span for each router request. Data can come from the request, the router configuration, or external sources. The example modifies the first span of the router but depending on the hook, the span is different.
+This module allows custom metric client to record request latency and error type.
 
 ```go
 type MyModule struct{}
 
-// Ensure that MyModule implements the RouterRequestHook interface
-var _ RouterRequestHook = (*MyModule)(nil)
+// Ensure that MyModule implements the RouterHook interface
+var _ RouterHook = (*MyModule)(nil)
+
+func (m *MyModule) PreRouterRequest(req *core.RouterRequest, err error) error {
+	tags := map[string]string{
+		"operation_name": req.OperationName,
+		"operation_type": req.OperationType,
+	}
+	m.scope.Tagged(tags).Counter("request.calls").Inc(1)
+
+	req.Context = context.WithValue(req.Context, startTimeKey, time.Now())
+
+	return nil
+}
+
+func (m *MyModule) OnRouterResponse(res *core.RouterResponse, err error) error {
```

Comment:
The use case would be covered but I'm still not happy how data is shared across hooks. I think it can become cumbersome, if you have to use the context api. It would be great to have an API to hook into pre/post without the need to share data.